packages/router/src/router.js

```diff
@@ -0,0 +1,88 @@
+// The guts of the router implementation.
+
+import {
+  Location,
+  parseSearch,
+  replaceParams,
+  matchPath,
+  ParamsContext,
+  navigate,
+  mapNamedRoutes,
+  SplashPage,
+} from './internal'
+
+// Definitions of the core param types.
+const coreParamTypes = {
+  Int: {
+    constraint: /\d+/,
+    transform: parseInt,
```

Comment:
`Number` might be more appropriate in this case because `parseInt` has a second `radix` argument that [isn't always base `10`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Description)

![image](https://user-images.githubusercontent.com/44849/73136638-f679e880-404f-11ea-997e-831a53e4a3f3.png)

🤦‍♂, ah software! `Number()` will do the trick here.

```js
Number("1")
// 1
Number("1.1")
// 1.1
Number("99999abc")
// NaN
Number("abc999")
//NaN
```

Reply:
Good call, I'll switch to use `Number()`.